In [1]:
import numpy as np
import scipy
import pandas as pd

# Q1

In [46]:
#Step1: Define hypothesis 
#x1= mean of unit A
#x2= mean of unit B
# Null hypothesis H0 : x1-x2=0
#Alternative hypothesisHa: x1-x2 !=0

In [47]:
#Step2 identify Z test or t test 

In [30]:
#Given Data

In [6]:
df= pd.read_csv(r"C:\Users\Mangesh-PC\Desktop\Excelr Class/cutlets.csv")

In [48]:
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [ ]:
#Here two samples are given. Hence We use two sample t test

In [49]:
#sample1
A=df.iloc[:,0]

In [50]:
#Sample2
B=df.iloc[:,1]

In [51]:
df.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [76]:
#Parameters
x1= 7.019091
x2= 6.964297
s1= 0.288408
s2= 0.343401
#dof= n1+n2-2
n1=35
n2=35
dof= 35+35-2
'\u03B1' # unicode for alpha
α=0.05

In [86]:
import sys
import unicodedata

In [77]:
t= (x1-x2)/ np.sqrt((s1*s1/n1)+(s2*s2/n2))

In [78]:
t

0.7228651675005472

In [41]:
x1-x2

0.05479400000000023

In [79]:
np.sqrt((s1**2+s2**2)/n1)

0.0758011347945587

In [80]:
0.05479400000000023/0.0758011347945587

0.7228651675005472

In [81]:
tcri = stats.t.ppf(0.05,68)

In [82]:
tcri

-1.6675722806611732

In [ ]:
# here t > tcri. Hence We conclude that we fail to reject null hypothesis

In [83]:
#2nd Method
#For 2 sample t test
from scipy import stats

In [27]:
T=stats.ttest_ind(A,B)

In [28]:
T

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

# Conclusion
    # Here p > α (0.4722 > 0.05). Hence we fail to reject null hypothesis
    # there is no significant difference in the diameter of the cutlet between two units

# Q2

In [84]:
#Step1: Define hypothesis 
#μ1= mean of Laboratory 1
#μ2= mean of Laboratory 2
#μ3= mean of Laboratory 3
#μ4= mean of Laboratory 4
# Null hypothesis H0 : μ1=μ2=μ3=μ4
#Alternative hypothesisHa: at least one sample is diff

In [89]:
'\u03BC'

'μ'

In [ ]:
#Loading data 

In [90]:
dfq2= pd.read_csv(r"C:\Users\Mangesh-PC\Desktop\Excelr Class/LabTAT.csv")

In [105]:
dfq2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [131]:
import scipy.stats as stats
# stats f_oneway functions takes the groups as input and returns ANOVA F and p value
fvalue, pvalue = stats.f_oneway(dfq2['Laboratory 1'],dfq2['Laboratory 2'],dfq2['Laboratory 3'],dfq2['Laboratory 4'])
print("F value is",fvalue,"and","P value is" ,pvalue)

F value is 118.70421654401437 and P value is 2.1156708949992414e-57


# Conclusion

#Here p value is less than 0.05. Hence we reject null hypothesis.
# This implies that TAT for at least one lab is diff

# Q3

In [ ]:
#Step1: Define hypothesis 
#H0: There is no diff in between male and female buyer ratio across all the regions
#Ha: There is  diff in between male and female buyer ratio at least in one region

#μe= mean of male and female buyer ratio for east regions
#μw= mean of male and female buyer ratio for west regions
#μn= mean of male and female buyer ratio for north regions
#μs= mean of male and female buyer ratio for south regions
# Null hypothesis H0 : μe=μs=μn=μw
#Alternative hypothesisHa: all means are not equal

In [ ]:
# loading Data

In [132]:
dfq3= pd.read_csv(r"C:\Users\Mangesh-PC\Desktop\Excelr Class/BuyerRatio.csv")

In [134]:
dfq3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [135]:
import scipy.stats as stats
# stats f_oneway functions takes the groups as input and returns ANOVA F and p value
fvalue, pvalue = stats.f_oneway(dfq3['East'],dfq3['West'],dfq3['North'],dfq3['South'])
print("F value is",fvalue,"and","P value is" ,pvalue)

F value is 0.3068565327605033 and P value is 0.8204060154887464


# Conclusion

#Here p value is greater than 0.05. Hence we fail to reject null hypothesis.

# This implies that there is no diff in between male and female buyer ratio across all the regions

# Q4

In [ ]:
#Step1: Define hypothesis 
#Assign
#Error free: 1
#Defective:0
#H0: The average number of error free customer order forms are equal across all the centers.
#Ha: The average number of error free customer order forms are diff for at least one centers.

In [15]:
#Data loading
dfq4= pd.read_csv(r"C:\Users\Mangesh-PC\Desktop\Excelr Class/Q4.csv")

In [16]:
dfq4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [17]:
dfq5 = pd.get_dummies(dfq4, columns=dfq4.columns )


In [18]:
dfq5.head()

Phillippines_Defective  Phillippines_Error Free  Indonesia_Defective  \
0                       0                        1                    0   
1                       0                        1                    0   
2                       0                        1                    1   
3                       0                        1                    0   
4                       0                        1                    0   

   Indonesia_Error Free  Malta_Defective  Malta_Error Free  India_Defective  \
0                     1                1                 0                0   
1                     1                0                 1                1   
2                     0                1                 0                0   
3                     1                0                 1                0   
4                     1                1                 0                0   

   India_Error Free  
0                 1  
1                 0  
2                 1  
3                 1  
4                 1

In [19]:
A1=dfq4["Phillippines"].replace({"Error Free": "1", "Defective": "0"})
B1=dfq4["Indonesia"].replace({"Error Free": "1", "Defective": "0"})
C1=dfq4["Malta"].replace({"Error Free": "1", "Defective": "0"})
D1=dfq4["India"].replace({"Error Free": "1", "Defective": "0"})

In [7]:
B1

0      1
1      1
2      0
3      1
4      1
      ..
295    1
296    1
297    1
298    1
299    0
Name: Indonesia, Length: 300, dtype: object

In [150]:
import scipy.stats as stats
# stats f_oneway functions takes the groups as input and returns ANOVA F and p value
fvalue, pvalue = stats.f_oneway(A1,B1,C1,D1)
print("F value is",fvalue,"and","P value is" ,pvalue)

F value is 1.286168556089167 and P value is 0.2776780955705948


# Conclusion

#Here p value is greater than 0.05. Hence we fail to reject null hypothesis.
# This implies that The average number of error free customer order forms are equal across all the centers